geom/FieldFunctions.F90

```diff
@@ -0,0 +1,85 @@
+#include "MAPL_Generic.h"
+
+module MAPL_FieldFunctions
+   use ESMF
+   use MAPL_ExceptionHandling
+   use MAPL_FieldPointerUtilities
+   use, intrinsic :: iso_fortran_env, only: REAL32, REAL64
+   implicit none
+   private
+
```

Comment:
In the MAP proposal I suggested a different approach that would rely less on templating.    Of course I did not have to take it as far, so may have missed something.   But @bena-nasa and @darianboggs  and myself should discuss to see if we want to aim in that direction instead.

Reply:
@tclune, I'm not sure what you mean, seems like this was about the minimum amount of code I could have done to implement this.

Reply:
I read the MAP proposal, section, now even more confused, to quote:
"The lowest layer will be a set of procedures that act on 1D arrays of data"
why would we need this? If I want to take the sin or cosine or any intrinsic fortran function of a 1D array, since they already intrinsically work on 1D arrays, why do we need such procedure, at some point you will get a 1D representation of the field, why would reimplement a "sin" procedure for example when Fortran's sin can already to a 1D array.